diff --git a/src/NodeFormAlter.php b/src/NodeFormAlter.php
index e22c063..3e1ffd0 100644
--- a/src/NodeFormAlter.php
+++ b/src/NodeFormAlter.php
@@ -216,7 +216,7 @@ class NodeFormAlter implements ContainerInjectionInterface {
     $group_menu_parent_options = $this->menuParentSelector
       ->getParentSelectOptions($defaults['id'], $group_menus);
 
-    $traditional_menu_parent_options = $form['menu']['link']['menu_parent']['#options'];
+    $traditional_menu_parent_options = $form['menu']['link']['menu_parent']['#options'] ?? [];
     $form['menu']['link']['menu_parent']['#options'] = $group_menu_parent_options + $traditional_menu_parent_options;
 
     if (!empty($has_link_in_group_menus)) {
